src/python/pants/backend/go/util_rules/build_pkg.py

```diff
@@ -892,6 +894,7 @@ async def build_go_package(
                 description=f"Compile Go package: {request.import_path}",
                 output_files=("__pkg__.a", *([asm_header_path] if asm_header_path else [])),
                 env={"__PANTS_GO_COMPILE_ACTION_ID": action_id_result.action_id},
+                replace_sandbox_root_in_args=True,
```

Comment:
I was curious to see how this works, and it's pretty straight forward:
https://github.com/pantsbuild/pants/blob/f61a9525b7af4e0316764688c9e5e7b169a4b5e8/src/python/pants/backend/go/util_rules/sdk.py#L158-L159
https://github.com/pantsbuild/pants/blob/f61a9525b7af4e0316764688c9e5e7b169a4b5e8/src/python/pants/backend/go/util_rules/sdk.py#L98-L103